File: tests/rust/roundtrips/line_strips3d/src/main.rs
//! Logs a `LineStrips3D` archetype for roundtrip checks.

use rerun::{RecordingStream, archetypes::LineStrips3D};

#[derive(Debug, clap::Parser)]
#[clap(author, version, about)]
struct Args {
    #[command(flatten)]
    rerun: rerun::clap::RerunArgs,
}

fn run(rec: &RecordingStream, _args: &Args) -> anyhow::Result<()> {
    let points = [[0., 0., 0.], [2., 1., -1.], [4., -1., 3.], [6., 0., 1.5]];
    rec.log(
        "line_strips3d",
        &LineStrips3D::new(points.chunks(2))
            .with_radii([0.42, 0.43])
            .with_colors([0xAA0000CC, 0x00BB00DD])
            .with_labels(["hello", "friend"])
            .with_class_ids([126, 127]),
    )
    .map_err(Into::into)
}

fn main() -> anyhow::Result<()> {
    re_log::setup_logging();

    use clap::Parser as _;
    let args = Args::parse();

    let (rec, _serve_guard) = args.rerun.init("rerun_example_roundtrip_line_strips3d")?;
    run(&rec, &args)
}

File: tests/cpp/roundtrips/line_strips3d/main.cpp
#include <rerun/archetypes/line_strips3d.hpp>
#include <rerun/recording_stream.hpp>

int main(int, char** argv) {
    const auto rec = rerun::RecordingStream("rerun_example_roundtrip_line_strips3d");
    rec.save(argv[1]).exit_on_failure();

    rec.log(
        "line_strips3d",
        rerun::archetypes::LineStrips3D(
            {rerun::components::LineStrip3D({{0.f, 0.f, 0.f}, {2.f, 1.f, -1.f}}),
             rerun::components::LineStrip3D({{4.f, -1.f, 3.f}, {6.f, 0.f, 1.5f}})}
        )
            .with_radii({0.42f, 0.43f})
            .with_colors({0xAA0000CC, 0x00BB00DD})
            .with_labels({"hello", "friend"})
            .with_class_ids({126, 127})
            .with_instance_keys({66, 666})
    );
}
